spec/javascripts/components/shared/icon_spec.js

```diff
@@ -0,0 +1,21 @@
+import jasmineEnzyme from 'jasmine-enzyme';
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Icon from 'components/shared/Icon';
+
+describe('<Icon />', function() {
+
+  beforeEach(jasmineEnzyme);
+
+  it("should be material icon", function() {
```

Comment:
Shouldn't you include the case here? Like `ìt("should be material icon when material prop is check_circle")`. Maybe you could create two different contexts to test this component. The first one with material is equals to `check_circle` and the second when the classname is equals to icon.